src/libstd/primitive_docs.rs

```diff
@@ -1036,6 +1040,11 @@ mod prim_ref { }
 /// [`FnMut`]: ops/trait.FnMut.html
 /// [`FnOnce`]: ops/trait.FnOnce.html
 ///
+/// Function pointers are pointers that point to *code*, not data. They can be called
```

Comment:
As @Lokathor said... are Lispers going to scream at this? ;)

Reply:
Is there any actionable feedback here?^^

Reply:
I thought we could say that they point to data that is code but that just detracts from the point that is made here... So the lispers will just have too live with resenting this bit. =P

Reply:
So, say I make an array of integers at runtime, then I take a pointer to that array and cast it to a pointer to a function and call the function.

The OS probably will object since it's not executable memory or whatever, but will _The Rust Language_ object? If there's no OS in the way can I do it? This is a meaningful question in the context of the limits of embedded development.

Reply:
@Lokathor congratulations, you found something that is *even harder* to [specify than `asm!`](https://github.com/rust-lang/rfcs/pull/2360#issuecomment-513463931). ;)

The Rust Abstract Machine does not have any way to support this. And if the compiler does global analyses, it might assume the program never does certain things even though your "JIT'ed" code does. But OTOH people do write JITs, so we can't make this UB... let's call this "unspecified behavior no really I mean it this is so far out there"?

Reply:
XD

I did really contemplate doing this once for the GBA as a way to get a complete game programmable in Stable, because there are some ASM calls you need to make the BIOS do things.

Reply:
This compiles on stable Rust, so if the Rust language isn't ok with it then somebody should open a soundness bug ([playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2015&gist=eb06d36618013287cb02bfa4a44c8ed4)). 

The OS isn't ok with this, but removing the `mut` in the `static mut` makes everything work as expected ([playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2015&gist=0e8b34da8aa258d678ddbc886d678c16)).